telemetry/aiwrapper_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Microsoft. All rights reserved.
+package telemetry
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-container-networking/aitelemetry"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateAITelemetryHandle(t *testing.T) {
+	tests := []struct {
+		name          string
+		aiConfig      aitelemetry.AIConfig
+		disableAll    bool
+		disableMetric bool
+		disableTrace  bool
+		wantErr       bool
+	}{
+		{
+			name:          "disable telemetry",
+			aiConfig:      aitelemetry.AIConfig{},
+			disableAll:    false,
+			disableMetric: true,
+			disableTrace:  true,
+			wantErr:       true,
+		},
+		{
+			name:          "empty aiconfig",
+			aiConfig:      aitelemetry.AIConfig{},
+			disableAll:    true,
+			disableMetric: true,
+			disableTrace:  true,
+			wantErr:       true,
+		},
+	}
+
+	for _, test := range tests {
+		tt := test
+		t.Run(tt.name, func(t *testing.T) {
+			err := CreateAITelemetryHandle(tt.aiConfig, tt.disableAll, tt.disableMetric, tt.disableTrace)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
```

Comment:
use require, it exits the test on a failed assertion
```suggestion
				require.Error(t, err)
```

Reply:
@rbtr  Used `assert` to keep iterating the rest of them even though one is failed.

Reply:
with table driven tests, the other scenarios will still be executed, this will just stop the current scenario from progressing. i think it makes sense to use it on the error/noerror because testing anything after that if you don't have what you expect doesn't make sense.